awstesting/assert.go

```diff
@@ -112,16 +114,24 @@ func AssertJSON(t *testing.T, expect, actual string, msgAndArgs ...interface{})
 }
 
 // AssertXML verifies that the expect xml string matches the actual.
-func AssertXML(t *testing.T, expect, actual string, container interface{}, msgAndArgs ...interface{}) bool {
-	expectVal := container
-	if err := xml.Unmarshal([]byte(expect), &expectVal); err != nil {
-		t.Errorf(errMsg("unable to parse expected XML", err, msgAndArgs...))
+// Elements in actual must match the order they appear in expect to match equally
+func AssertXML(t *testing.T, expect, actual string, msgAndArgs ...interface{}) bool {
+	buffer := bytes.NewBuffer([]byte(expect))
```

Comment:
Minor issue, do you mind using a `bytes.Reader` here instead of `Buffer` since the buffer is never written to, and only read.